Complete usage documentation

MonsieurV opened the pull request:
Hi @artemeff,

I fixed some bits on the usage documentation, so users can get started more rapidly.

Thanks for the lib, it works great.

MonsieurV committed:
Complete documentation.

Add Hex status
Add sample Rebar and app.src config files
Tell ipfamily parameter is optional

stolyaroleh commented on README.md:
````diff
@@ -24,7 +49,7 @@ It will accept either the individual config components:
 
 ```erlang
-{raven, [
+{raven_erlang, [
````
It might be a good idea to add `ipfamily` here too, since it is [required](https://github.com/artemeff/raven-erlang/blob/master/src/raven.erl#L100) in both cases.

MonsieurV replied:
That's strange, I remember using it without any `ipfamily` entry in my configuration:
```erl
    {lager, [
        {handlers, [
            {lager_console_backend, [{level, info}]},
            {raven_lager_backend, error}
      ]}
    ]},
    {raven_erlang, [
        {dsn, "https://<uid1>:<uid2>@sentry.io/<id>"},
        {error_logger, false}
    ]}
].
```

It was working.

Was I missing something? Does it default to `inet` if no provided?

No problem to add it to the PR if it is indeed needed

stolyaroleh replied:
It will fail to match on `{ok, IpFamily}` if it's not provided - see link above to `get_config`. I tested it yesterday. Maybe it's a recent change.

MonsieurV replied:
Indeed!

Added it, thx :)

artemeff replied:
we can make it optional and use inet as default value for ipfamily

MonsieurV replied:
:+1:  for that. This will let use the library by just specifying the DSN.

I made the change and documented `ipfamily` as optional. Should I include in the PR the code to actually render it optional or you take care of that?

artemeff replied:
@MonsieurV include, please :)

like `IpFamily = application:get_env(App, ipfamily, inet)` [here](https://github.com/artemeff/raven-erlang/blob/master/src/raven.erl#L100)

and then please squash commits into two — first for docs and second about ipfamily config change :)

MonsieurV replied:
Ok, done it! :)

Is that good @artemeff?

I saw a default value was already provided for `ipfamily` in:

https://github.com/artemeff/raven-erlang/blob/170f036d7a8125416b3c257854735df16548c85f/src/raven_erlang.app.src#L9

I anyway added an explicit default value in the code using `application:get_env()` as you suggested.

artemeff commented on README.md:
````diff
@@ -1,17 +1,42 @@
 # raven-erlang
 
-raven-erlang is an Erlang client for [Sentry](http://aboutsentry.com/) that integrates with the standard ```error_logger``` module.
+raven-erlang is an Erlang client for [Sentry](http://aboutsentry.com/) that integrates with the standard `error_logger` module. It also serves as a [Lager](https://github.com/basho/lager) backend.
````
lager lives at https://github.com/erlang-lager/lager

MonsieurV committed:
Explicitely make ipfamily optional and default to inet

Was already defined in raven_erlang.app.src as inet if not overriden

artemeff commented on src/raven.erl:
```diff
@@ -97,7 +97,7 @@ get_config() ->
 -spec get_config(App :: atom()) -> cfg_rec().
 get_config(App) ->
-	{ok, IpFamily} = application:get_env(App, ipfamily),
+	{ok, IpFamily} = application:get_env(App, ipfamily, inet),
```
`get_env(Application, Par, Def) -> Val`, not `{ok, Val}` :)

MonsieurV committed:
Add tests for application configuration loading and for a simple event capture.

This make clear the bug I introduced earlier with {ok, IpFamily} =
application:get_env(App, ipfamily, inet)

Also add Travis CI config

MonsieurV committed:
Actually fix the bug I introduced earlier: get_env() with a default specified directly return the value (and not the tuple {ok, Value})

MonsieurV commented:
@artemeff fortunately you review what I do!

When I committed the change, I thought:
> Should I test this? Seems simple enough. I can't do harm there.

Well, proof that it should _not_ be a question! ahah

So this time I added tests, because I don't trust myself for not breaking things :smile:

artemeff commented:
Woah, few years I'm waiting for a tests! You are awesome! 👍
